server/api/oauthEnabledClient.js

```diff
@@ -1,7 +1,24 @@
 const axios = require('axios')
+const superagent = require('superagent')
 const { logger } = require('../services/logger')
 
-const { addAuthorizationHeader, addPaginationHeaders } = require('./axios-config-decorators')
+const { addAuthorizationHeader, addPaginationHeaders, getHeaders } = require('./axios-config-decorators')
+
+const superAgentResultHandler = result => {
+  logger.debug(`${result.req.method} ${result.req.path} ${result.status}`)
+  return result
+}
+
+const superAgentErrorHandler = error => {
+  const status = error.response ? error.response.status : '-'
+  const responseData = error.response ? error.response.data : '-'
+  if (error.response && error.response.req) {
+    logger.warn(
+      `API error in ${error.response.req.method} ${error.response.req.path} ${status} ${error.message} ${responseData}`
+    )
+  } else logger.warn(`API error with message ${error.message}`)
+  return error
```

Comment:
superagent needs the error to be returned rather than thrown here.  Eventually will be able to replace the loggers with the handlers